docs/admin/optimization.rst

```diff
@@ -15,21 +15,27 @@ Introduction
 
 In CrateDB every table (or if partitioned every partition) consists of
 segments. When inserting/deleting/updating data new segments are created
-following as an append-only strategy, which gives the advantage of fast writes
-but on the other hand can result into a big number of segments. As the number
+following an append-only strategy, which gives the advantage of fast writes
+but on the other hand can result in a large number of segments. As the number
 of segments increases the read operations become slower since more segments
 need to be visited. Moreover each segment consumes file handles, memory and
 CPU. CrateDB solves this problem by merging segments automatically in the
 background. Small segments are merged into bigger segments, which, in turn, are
-merged into even bigger segments. Furthermore any deleted rows and documents
-are not copied to the new bigger segment during this process.
+merged into even bigger segments. Furthermore any deleted rows
+are not copied to the new bigger segment during this process, and recovery
+source and soft-delete markers which are necessary for peer-recovery can be
+removed for rows that have been fully replicated.
 
 If required one or more tables or table partitions can be optimized explicitly
 in order to improve performance. A few parameters can also be configured for
 the optimization process, like the max number of segments you wish to have when
 optimization is completed, or if you only wish to merge segments with deleted
 data, etc. See :ref:`sql-optimize` for detailed description of parameters.
 
+Disk usage after an ``OPTIMIZE`` call should be lower than it was before,
+but the optimization process itself may temporarily use up to twice as much disk
+space as the initial size of the table or partition.
```

Comment:
Can we narrow down this a bit? maybe based on some expected level of parallelism to the size of a number of shards? asking as it has implications for sizing.

Reply:
It's tricky to be more specific as it will depend on the segment geometry.  This is from the [lucene javadocs for `forceMerge(int)`](https://lucene.apache.org/core/9_12_0/core/org/apache/lucene/index/IndexWriter.html#forceMerge(int)):

> Note that this requires free space that is proportional to the size of the index in your Directory: 2X if you are not using compound file format, and 3X if you are. For example, if your index size is 10 MB then you need an additional 20 MB free for this to complete (30 MB if you're using compound file format). This is also affected by the [Codec](https://lucene.apache.org/core/9_12_0/core/org/apache/lucene/codecs/Codec.html) that is used to execute the merge, and may result in even a bigger index. Also, it's best to call [commit()](https://lucene.apache.org/core/9_12_0/core/org/apache/lucene/index/IndexWriter.html#commit()) afterwards, to allow IndexWriter to free up disk space.

> If some but not all readers re-open while merging is underway, this will cause > 2X temporary space to be consumed as those new readers will then hold open the temporary segments at that time. It is best not to re-open readers while merging is running.

> The actual temporary usage could be much less than these figures (it depends on many factors).